fe/src/main/java/org/apache/doris/external/elasticsearch/EsIndexState.java

```diff
@@ -44,7 +47,43 @@ public EsIndexState(String indexName) {
         this.partitionDesc = null;
         this.partitionKey = null;
     }
-
+    
+    /**
+     * Parse shardRoutings from the json
+     * @param indexName indexName(alias or really name)
+     * @param shardLocation the return value of _search_shards
+     * @return shardRoutings is used for searching
+     */
+    public static EsIndexState fromShardLocation(String indexName, String shardLocation) {
```

Comment:
```suggestion
    public static EsShardPartitions findShardPartitions(String indexName, String shardLocation) {
```

Reply:
Doris On ES maybe would use `slice scroll` in future,  at this time, we would add `findSlicePartitions`